packages/services/src/kernel/default.ts

```diff
@@ -235,14 +246,10 @@ class DefaultKernel implements Kernel.IKernel {
     this._terminated.emit(void 0);
     this._status = 'dead';
     this._clearSocket();
-    this._futures.forEach((future, key) => {
-      future.dispose();
-    });
-    this._commPromises.forEach((promise, key) => {
-      promise.then(comm => {
-        comm.dispose();
-      });
-    });
+    this._futures.forEach(future => { future.dispose(); });
+    this._comms.forEach(comm => { comm.dispose(); });
+    this._kernelSession = '';
+    this._msgChain = null;
```

Comment:
How close are we to enabling strict null checks? Should we type `msgChain` as `Promise<void> | null`?

Reply:
done.